packages/phpstan-rules/src/Rules/PreventDuplicateClassMethodRule.php

```diff
@@ -0,0 +1,192 @@
+ */
+final class PreventDuplicateClassMethodRule extends AbstractSymplifyRule
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = 'Content of method "%s" is duplicated with method in "%s" class, use unique content instead';
+
+    /**
+     * @var string[]
+     */
+    private const PHPSTAN_GET_NODE_TYPE_METHODS = ['getNodeType', 'getNodeTypes'];
+
+    /**
+     * @var SimpleNameResolver
+     */
+    private $simpleNameResolver;
+
+    /**
+     * @var Standard
+     */
+    private $printerStandard;
+
+    /**
+     * @var array<string, string>
+     */
+    private $firstClassByName = [];
+
+    /**
+     * @var array<string, string>
+     */
+    private $contentMethodByName = [];
+
+    public function __construct(SimpleNameResolver $simpleNameResolver, Standard $printerStandard)
+    {
+        $this->simpleNameResolver = $simpleNameResolver;
+        $this->printerStandard = $printerStandard;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     * @return string[]
+     */
+    public function process(Node $node, Scope $scope): array
+    {
+        /** @var Class_|null */
+        $class = self::resolveCurrentClassWithNamespacedNameCheck($node);
+
+        if ($class === null) {
+            return [];
+        }
+
+        $className = (string) $class->namespacedName;
+        if ($this->isConstructorOrInTestClass($node, $className)) {
+            return [];
+        }
+
+        $classMethodName = (string) $node->name;
+        if (in_array($classMethodName, self::PHPSTAN_GET_NODE_TYPE_METHODS, true)) {
+            return [];
+        }
+
+        if (! $node->isPublic()) {
+            return [];
+        }
+
+        /** @var Node[] $stmts */
+        $stmts = $node->stmts;
+        if (count($stmts) === 1 && ($stmts[0] instanceof Return_ || $stmts[0] instanceof Expression)) {
+            return [];
+        }
+
+        $printStmts = $this->printerStandard->prettyPrint($stmts);
+
+        if (! isset($this->contentMethodByName[$classMethodName])) {
+            $this->firstClassByName[$classMethodName] = $className;
+            $this->contentMethodByName[$classMethodName] = $printStmts;
+            return [];
+        }
+
+        if ($printStmts !== $this->contentMethodByName[$classMethodName]) {
+            return [];
+        }
+
+        return [sprintf(self::ERROR_MESSAGE, $classMethodName, $this->firstClassByName[$classMethodName])];
+    }
+
+    private function resolveCurrentClassWithNamespacedNameCheck(ClassMethod $classMethod): ?Class_
+    {
+        /** @var Class_|null */
+        $class = $this->resolveCurrentClass($classMethod);
+
+        if ($class === null) {
+            return null;
+        }
+
+        if (! property_exists($class, 'namespacedName')) {
+            return null;
+        }
+
+        return $class;
+    }
```

Comment:
Here, the parent method could be used instead: `\Symplify\PHPStanRules\Rules\AbstractSymplifyRule::getClassName()`

Reply:
implemented.